networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/topics/GossipTopics.java

```diff
@@ -65,19 +64,19 @@ public static String getBlobSidecarSubnetTopic(
   }
 
   public static Set<String> getAllTopics(
-      final GossipEncoding gossipEncoding,
-      final Bytes4 forkDigest,
-      final NetworkingSpecConfig networkingConfig) {
+      final GossipEncoding gossipEncoding, final Bytes4 forkDigest, final Spec spec) {
     final Set<String> topics = new HashSet<>();
 
-    for (int i = 0; i < networkingConfig.getAttestationSubnetCount(); i++) {
+    for (int i = 0; i < spec.getNetworkingConfig().getAttestationSubnetCount(); i++) {
       topics.add(getAttestationSubnetTopic(forkDigest, i, gossipEncoding));
     }
     for (int i = 0; i < NetworkConstants.SYNC_COMMITTEE_SUBNET_COUNT; i++) {
       topics.add(getSyncCommitteeSubnetTopic(forkDigest, i, gossipEncoding));
     }
-    for (int i = 0; i < Constants.BLOB_SIDECAR_SUBNET_COUNT; i++) {
-      topics.add(getBlobSidecarSubnetTopic(forkDigest, i, gossipEncoding));
+    if (spec.getNetworkingConfigDeneb().isPresent()) {
```

Comment:
We can use something like `spec.getNetworkingConfig().toVersionDeneb().ifPresent(networkingConfigDeneb ->.....)` potentially?

Reply:
We could refactor it in function style but I don't think it looks less ugly
https://github.com/Consensys/teku/pull/7304/commits/da26990115c042cd349140040f5b326447ff58ce